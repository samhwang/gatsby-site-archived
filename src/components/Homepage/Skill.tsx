import React from 'react';

function Skill(props: { title: string; description: string[] }) {
  const { title, description } = props;
  return (
    <>
      <div className="subheading mb-3">{title}</div>
      <p>
        <ul>
          {description.map((line, key) => (
            <li key={key}>{line}</li>
          ))}
        </ul>
      </p>
    </>
  );
}

export default Skill;
