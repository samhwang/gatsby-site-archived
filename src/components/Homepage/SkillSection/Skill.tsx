export interface SkillProps {
  title: string;
  description: string[];
}

function Skill({ title, description }: SkillProps) {
  return (
    <>
      <div className="subheading mb-3">{title}</div>
      <ul>
        {description.map((line) => (
          <li key={line}>{line}</li>
        ))}
      </ul>
    </>
  );
}

export default Skill;
