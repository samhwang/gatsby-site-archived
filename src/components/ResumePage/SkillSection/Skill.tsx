import type { FC } from 'react';

export interface SkillProps {
  title: string;
  description: string[];
}

const Skill: FC<SkillProps> = ({ title, description }) => (
  <>
    <div className="subheading mb-3">{title}</div>
    <ul>
      {description.map((line) => (
        <li key={line}>{line}</li>
      ))}
    </ul>
  </>
);

export default Skill;
