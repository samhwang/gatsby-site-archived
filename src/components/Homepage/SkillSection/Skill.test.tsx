/* eslint-disable import/no-extraneous-dependencies */
import { render } from '@testing-library/react';
import Skill from './Skill';

describe('Skill Component', () => {
  it('Should render without crashing and match snapshot', () => {
    const skill = {
      title: 'asdfasdf',
      description: ['asdfasdf1234'],
    };
    const tree = render(
      <Skill title={skill.title} description={skill.description} />
    );
    expect(tree).toMatchSnapshot();
  });
});
