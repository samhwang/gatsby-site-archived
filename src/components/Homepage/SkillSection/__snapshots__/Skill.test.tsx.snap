// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skill Component Should render without crashing and match snapshot 1`] = `
Array [
  <div
    className="subheading mb-3"
  >
    asdfasdf
  </div>,
  <ul>
    <li>
      asdfasdf1234
    </li>
  </ul>,
]
`;
